snuba/datasets/spans_processor.py

```diff
@@ -77,6 +77,15 @@ def __fill_common(
         tags = _as_dict_safe(data.get("tags", None))
         span["tags.key"], span["tags.value"] = extract_extra_tags(tags)
 
+    def __safe_extract_int(self, field: str, value: Any) -> Optional[int]:
+        if value is not None and isinstance(value, str):
+            return int(value, 16)
+        elif value is not None:
```

Comment:
What if the value is already an int?

Reply:
Good point, technically it should still be invalid, but it could be stored that way in a span. Updated.